Data_Structures_Answers.md

```diff
@@ -1,18 +1,28 @@
 Add your answers to the questions below.
+O(1) describes an algorithm that will always execute in the same time (or space) regardless of the size of the input data set.
+O(N) describes an algorithm whose performance will grow linearly and in direct proportion to the size of the input data set.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+O(1) using the array data structure
 
 2. What is the space complexity of your ring buffer's `append` function?
+and for the space it will be O(n) since we inserting in an array
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+ O(n)
```

Comment:
The solution implementation returns a new list with the elements in storage, i.e. it needs to add each of the elements to the new list one at a time. Thus O(n) in the worst case. However, it is entirely possible to implement this method such that it exhibits O(1) time complexity.